src/modules/Trip/test/trip.test.js

```diff
@@ -0,0 +1,81 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../../index';
+
+chai.should();
+chai.use(chaiHttp);
+describe('Testing for one-way trip request endpoint', () => {
+  it('user can make trip request if he is logged in and has a valid token', done => {
+    (async () => {
+      const validInput = {
+        email: 'admin@nomad.com',
+        password: '123456'
+      };
+      const data = {
+        origin: 'Lagos',
+        destination: 'Abj',
+        reason: 'kckcmlcm',
+        departureDate: '52555353553',
+        accomodationId: '1',
+        type: 'one-way',
+        returnDate: '6363ydd'
+      };
+      const request = chai.request(app).keepOpen();
+      const signResponse = await request.post('/api/v1/auth/signin').send(validInput);
+      const { token } = signResponse.body.data;
+      const res = await request.post('/api/v1/trip/request').set('Authorization', `Bearer ${token}`).send(data);
+      res.body.should.have.status('success');
+      done();
+    })();
+  });
+
+  it('user cannot make a trip request if token is empty', done => {
+    (async () => {
+      const validInput = {
+        email: 'admin@nomad.com',
+        password: '123456'
+      };
+      const data = {
+        origin: 'Lagos',
+        destination: 'Abj',
+        reason: 'kckcmlcm',
+        departureDate: '52555353553',
+        accomodationId: '1',
+        type: 'one-way',
+        returnDate: '12-12-2019'
+      };
+      const request = chai.request(app).keepOpen();
+      await request.post('/api/v1/auth/signin').send(validInput);
+      const token = '';
+      const res = await request.post('/api/v1/trip/request').set('Authorization', `Bearer ${token}`).send(data);
+      res.body.should.have.status('error');
+      res.should.have.status(403);
+      done();
+    })();
+  });
+
+  it('user cannot make a trip request if token is invalid', done => {
+    (async () => {
+      const validInput = {
+        email: 'admin@nomad.com',
+        password: '123456'
+      };
+      const data = {
+        origin: 'Lagos',
+        destination: 'Abj',
+        reason: 'kckcmlcm',
+        departureDate: '52555353553',
+        accomodationId: '1',
+        type: 'one-way',
+        returnDate: '12-12-2019'
+      };
+      const request = chai.request(app).keepOpen();
+      await request.post('/api/v1/auth/signin').send(validInput);
+      const token = 'dndkcndckdcnlsmlslmcksncjnscnnc';
+      const res = await request.post('/api/v1/trip/request').set('Authorization', `Bearer ${token}`).send(data);
+      res.body.should.have.status('error');
+      res.should.have.status(403);
+      done();
+    })();
+  });
+});
```

Comment:
I think there should be at least, some tests for the validation of form input as well.

Reply:
we are testing the endpoint not validation. i think basically you just need to verify if user is logged in , if user supply the token as a part of header, and also authenticate if the token is valid. in  a situation where you have ten or more  form fields will go ahead and validate each of the fields?

Reply:
Hmmm! But don't you think validating user input is part of testing the endpoint? if there is a validation middleware, I think it has to be tested.

Reply:
i think is more more of validation of claims, and endpoints. i think the form fields are somewhat trivia

Reply:
@solomonfrank, I just want to chip this in. Form validation is the most important aspect of this project because if you don't handle it well then anything can happen. so you need to test your form validation very well with as many use cases you can think of.